code/code/misc/discipline.cc

```diff
@@ -1359,8 +1360,91 @@ byte defaultProficiency(byte uLearned, byte uStart, byte uLearn)
   return((uLearned-uStart)*uLearn);
 }
 
+// CritSuccess for spells only
+// The goal here is a crit rate between
+// min - 0.1% -   10 out of 10000
+// max - 10%  - 1000 out of 10000
+// avg - 1%   -  100 out of 10000
+//
+// Factors should be
+// Wisdom:      Primary
+// Karma:       Secondary
+// Learnedness: Modifier
+// Position:    Modifier
+//
+critSuccT spellCritSuccess(TBeing *caster, spellNumT spell)
+{
+  // Base chance is 100
+  // If we are totally average then this will hold up
+  int chance = 100;
```

Comment:
As there's a considerable number of floating-point calculates involved, perhaps we'd want to keep this variable a float as well?

Reply:
You right. I changed it.